fastlane_bot/helpers/routehandler.py

```diff
@@ -658,6 +753,73 @@ def _extract_flashloan_tokens(self, trade_instructions: List[TradeInstruction])
 
         return flash_tokens
 
+    def split_carbon_trades(self, trade_instructions: List[TradeInstruction]) -> List[TradeInstruction]:
```

Comment:
Similar to `add_wrap_or_unwrap_trades_to_route_v4`, the `split_carbon_trades` function could also be refactored into a class following the same principles. Here is a quick and dirty refactor of the `split_carbon_trades` function into a class `CarbonTradeSplitter`:

```python
class CarbonTradeSplitter:
    NATIVE = 'native'
    WRAPPED = 'wrapped'

    def __init__(self, ConfigObj):
        self.ConfigObj = ConfigObj

    def split_carbon_trades(self, trade_instructions: List[TradeInstruction]) -> List[TradeInstruction]:
        """
        Split Carbon trades based on gas token type or exchange differences.
        """
        new_trade_list = []
        for trade in trade_instructions:
            if not self._is_carbon_trade(trade):
                new_trade_list.append(trade)
                continue

            carbon_exchanges = self._process_carbon_trades(trade)
            self._create_new_trades_from_carbon_exchanges(carbon_exchanges, trade, new_trade_list)

        return new_trade_list

    def _is_carbon_trade(self, trade: TradeInstruction) -> bool:
        """
        Check if a trade is a Carbon trade.
        """
        return trade.exchange_name in self.ConfigObj.CARBON_V1_FORKS

    def _process_carbon_trades(self, trade: TradeInstruction) -> Dict:
        """
        Process Carbon trades and organize them by exchange and token type.
        """
        carbon_exchanges = {}
        raw_txs = json.loads(trade.raw_txs)

        for _tx in raw_txs:
            curve = trade.db.get_pool(cid=str(_tx["cid"]).split('-')[0])
            exchange = curve.exchange_name
            token_type = self._get_token_type(curve, trade)

            if exchange not in carbon_exchanges:
                carbon_exchanges[exchange] = self._initialize_exchange_data()

            self._update_exchange_data(carbon_exchanges[exchange], token_type, _tx, trade)

        return carbon_exchanges

    def _get_token_type(self, curve, trade) -> str:
        """
        Determine if the trade uses native or wrapped gas tokens.
        """
        if self.ConfigObj.NATIVE_GAS_TOKEN_ADDRESS in curve.get_tokens:
            return self.NATIVE
        return self.WRAPPED

    def _initialize_exchange_data(self) -> Dict:
        """
        Initialize data structure for a new exchange.
        """
        return {
            self.NATIVE: self._new_trade_data_structure(),
            self.WRAPPED: self._new_trade_data_structure()
        }

    def _new_trade_data_structure(self) -> Dict:
        """
        Initialize a new trade data structure.
        """
        return {
            "raw_txs": [],
            "amt_in": 0,
            "amt_out": 0,
            "amtin_wei": 0,
            "amtout_wei": 0,
        }

    def _update_exchange_data(self, exchange_data: Dict, token_type: str, _tx: Dict, trade: TradeInstruction):
        """
        Update exchange data with information from a transaction.
        """
        data = exchange_data[token_type]
        data["raw_txs"].append(_tx)
        data["amt_in"] += _tx["amtin"]
        data["amt_out"] += _tx["amtout"]
        data["amtin_wei"] += _tx["_amtin_wei"]
        data["amtout_wei"] += _tx["_amtout_wei"]

        # Assuming tknin and tknout logic needs refinement based on actual requirements
        data["tknin"] = trade.tknin
        data["tknout"] = trade.tknout

    def _create_new_trades_from_carbon_exchanges(self, carbon_exchanges: Dict, original_trade: TradeInstruction, new_trade_list: List[TradeInstruction]):
        """
        Create new TradeInstruction instances from processed Carbon exchanges data.
        """
        for exchange, data in carbon_exchanges.items():
            for token_type, trade_data in data.items():
                if trade_data["raw_txs"]:
                    trade_data['cid'] = original_trade.cid
                    trade_data['db'] = original_trade.db
                    trade_data['ConfigObj'] = original_trade.ConfigObj
                    new_trade_list.append(TradeInstruction(**trade_data))
```